smithy-linters/src/main/java/software/amazon/smithy/linters/NoninclusiveTermsValidator.java

```diff
@@ -130,70 +135,79 @@ private Collection<ValidationEvent> getValidationEvents(TextInstance instance) {
             final int startIndex = instance.getText().toLowerCase().indexOf(termLower);
             if (startIndex != -1) {
                 final String matchedText = instance.getText().substring(startIndex, startIndex + termLower.length());
-                switch (instance.getLocationType()) {
-                    case NAMESPACE:
-                        //Cannot use any warning() overloads because there is no shape associated with the event.
-                        events.add(ValidationEvent.builder()
-                                .sourceLocation(SourceLocation.none())
-                                .id(this.getClass().getSimpleName().replaceFirst("Validator$", ""))
-                                .severity(Severity.WARNING)
-                                .message(formatNonInclusiveTermsValidationMessage(termEntry, matchedText, instance))
-                                .build());
-                        break;
-                    case APPLIED_TRAIT:
-                        events.add(warning(instance.getShape(),
-                                instance.getTrait().getSourceLocation(),
-                                formatNonInclusiveTermsValidationMessage(termEntry, matchedText, instance)));
-                        break;
-                    case SHAPE:
-                    default:
-                        events.add(warning(instance.getShape(),
-                                instance.getShape().getSourceLocation(),
-                                formatNonInclusiveTermsValidationMessage(termEntry, matchedText, instance)));
-                }
+                events.add(constructValidationEvent(instance, termEntry.getValue(), matchedText));
             }
         }
         return events;
     }
 
-    private static String formatNonInclusiveTermsValidationMessage(
-            Map.Entry<String, List<String>> termEntry,
-            String matchedText,
-            TextInstance instance
+    /**
+     * Creates a single ValidationEvent.
```

Comment:
Nit, missing a new line after first paragraph. Also, javadoc comments are usually for public API, not for private methods, so I'd rather remove it.

Reply:
Removed in next commit